lib/shipit/line_buffer.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Shipit
+  class LineBuffer
+    SEPARATOR = "\n"
+
+    def initialize(value = "")
+      @value = value
+    end
+
+    def buffer(text, &block)
+      @value += text
+      whole_lines.each(&block).tap { flush }
+    end
+
+    private
+
+    def whole_lines
+      whole? ? lines : lines[0..-2]
+    end
+
+    def flush
+      whole? ? clear : @value = lines.last
+    end
+
+    def whole?
+      @value.end_with?(SEPARATOR)
+    end
+
+    def lines
+      @value.split(SEPARATOR).reject(&:empty?)
```

Comment:
What's the reason for skipping empty lines? They might be somewhat meaningful.

Reply:
I didn't see the point in logging blank lines, but I can remove this if you think they would be meaningful.

Reply:
> I didn't see the point in logging blank lines

They usually improve readability. I'd rather have that logger as dumb as possible, as in it should chose what or what not to log.